Marlin/src/lcd/menu/menu_bed_corners.cpp

```diff
@@ -208,9 +208,15 @@ static void _lcd_level_bed_corners_get_next_position() {
     if (!ui.should_draw()) return;
     MenuItem_confirm::select_screen(
         GET_TEXT_F(TERN(HAS_LEVELING, MSG_BUTTON_LEVEL, MSG_BUTTON_DONE)),
-        TERN(HAS_LEVELING, GET_TEXT_F(MSG_BUTTON_BACK), nullptr)
-      , []{ queue.inject(TERN(HAS_LEVELING, F("G29N"), FPSTR(G28_STR))); ui.return_to_status(); }
-      , TERN(HAS_LEVELING, ui.goto_previous_screen_no_defer, []{})
+        TERN(HAS_LEVELING, GET_TEXT_F(MSG_BUTTON_DONE), nullptr)
+      , []{ TERN(HAS_LEVELING, []{queue.inject(F("G29N")); corner_probing_done = true;}, queue.inject(FPSTR(G28_STR))); ui.goto_previous_screen_no_defer();}
```

Comment:
- The code wrapped in `[]{}` will not be executed. It just defines an anonymous lambda.
- This screen is the final one, and it should go back to Status so leveling / homing are reported.